ShaderConverter/ShaderConv/context.cpp

```diff
@@ -1342,35 +1384,42 @@ CContext::Translate_NRM( const CInstr& instr )
 {
     // dp3  s0.x, src0, src0
     // rsq  s0.y, s0.x
-    // movc s0.z, s0.x, s0.y, vec4(FLT_MAX)
+    // mov  s0.z, abs( s0.x )
+    // movc s0.z, s0.z, s0.y, vec4( FLT_MAX, FLT_MAX, FLT_MAX, FLT_MAX )
     // mul  dest, src0, s0.z
 
     const COperandBase dest = instr.CreateDstOperand();
     const COperandBase src0 = this->EmitSrcOperand( instr, 0 );
 
     m_pShaderAsm->EmitInstruction(
         CInstruction( D3D10_SB_OPCODE_DP3,
-                      CTempOperandDst( SREG_TMP0, D3D10_SB_OPERAND_4_COMPONENT_MASK_X ),
-                      src0,
-                      src0 ) );
+            CTempOperandDst( SREG_TMP0, D3D10_SB_OPERAND_4_COMPONENT_MASK_X ),
+            src0,
+            src0 ) );
 
     m_pShaderAsm->EmitInstruction(
         CInstruction( D3D10_SB_OPCODE_RSQ,
-                      CTempOperandDst( SREG_TMP0, D3D10_SB_OPERAND_4_COMPONENT_MASK_Y ),
-                      CTempOperand4( SREG_TMP0, __SWIZZLE_X ) ) );
+            CTempOperandDst( SREG_TMP0, D3D10_SB_OPERAND_4_COMPONENT_MASK_Y ),
+            CTempOperand4( SREG_TMP0, __SWIZZLE_X ) ) );
 
     m_pShaderAsm->EmitInstruction(
-        CInstruction(   D3D10_SB_OPCODE_MOVC,
-                        CTempOperandDst(SREG_TMP0, D3D10_SB_OPERAND_4_COMPONENT_MASK_Z),
-                        CTempOperand4(SREG_TMP0, __SWIZZLE_X),
-                        CTempOperand4(SREG_TMP0, __SWIZZLE_Y),
-                        COperand( FLT_MAX, FLT_MAX, FLT_MAX, FLT_MAX ) ) );
+        CInstruction( D3D10_SB_OPCODE_MOV,
+            CTempOperandDst( SREG_TMP0, D3D10_SB_OPERAND_4_COMPONENT_MASK_Z ),
+            CAbs( CTempOperand4( SREG_TMP0, __SWIZZLE_X ) ) ) );
+
+    m_pShaderAsm->EmitInstruction(
+        CInstruction( D3D10_SB_OPCODE_MOVC,
+            CTempOperandDst( SREG_TMP0, D3D10_SB_OPERAND_4_COMPONENT_MASK_Z ),
+            CTempOperand4( SREG_TMP0, __SWIZZLE_Z ),
+            CTempOperand4( SREG_TMP0, __SWIZZLE_Y ),
+            COperand( FLT_MAX, FLT_MAX, FLT_MAX, FLT_MAX ) ) );
 
     this->EmitDstInstruction( instr.GetModifiers(),
                               D3D10_SB_OPCODE_MUL,
                               dest,
                               src0,
                               CTempOperand4( SREG_TMP0, __SWIZZLE_Z ) );
+

```

Comment:
The `mov  s0.z, abs( s0.x )` is a redundant op as S0.x must be positive (since it's the sum of squaring the components of src0)